### PR TITLE
[Priest] Unholy Nova

### PR DESCRIPTION
**Pull request by TJMoats**
This adds support for all three priest classes, but it doesn't do anything special for disc. Someone may want to go in and add some extra work to determine additional healing from atonement. 

Screen Shot 2020-12-28 at 10 52 39 AM

### CODE REVIEW

File: src/parser/priest/holy/CombatLogParser.tsx [hunk 1]
```diff
@@ -139,6 +142,9 @@ class CombatLogParser extends CoreCombatLogParser {
 
     // Conduits
     resonantWords: ResonantWords,
+
+    // Conduits
```
Comment (DylanDirlam) [inline src/parser/priest/holy/CombatLogParser.tsx]: Comment should be Covenants here for consistency

File: src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          didHit = true;
+          break;
+        }
+      }
+      if (!didHit) {
+        totalMisses += 1;
+      }
+    }
+    return totalMisses;
+  }
```
Comment (DylanDirlam) [inline src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx:54]: A suggestion to better place/time their casts would be good to have if they are missing their hits.

File: src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    super(options);
+
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.NECROLORD.id);
+    if (!this.active) {
+      return;
+    }
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.UNHOLY_NOVA), this.onCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.UNHOLY_TRANSFUSION_DAMAGE), this.onDamage);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.UNHOLY_TRANSFUSION), this.onHeal);
```
Comment (DylanDirlam) [inline src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx:67]: This heal event only includes the heal over time. An event listener for `UNHOLY_NOVA_BUFF` for the initial hit should be added.

Log for reference: https://www.warcraftlogs.com/reports/JAPL1zpDfN7W8wck#fight=26&type=healing&source=19

**Review by DylanDirlam — CHANGES_REQUESTED** [review]

File: src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx (hunk 2 above)
Comment (TJMoats) [reply src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx:54]: Screen+Shot+2020-12-28+at+12 18 38+PM

Suggestion added.

File: src/parser/priest/holy/CombatLogParser.tsx (hunk 1 above)
Comment (TJMoats) [reply src/parser/priest/holy/CombatLogParser.tsx]: Fixed

File: src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx (hunk 3 above)
Comment (TJMoats) [reply src/parser/priest/shared/modules/shadowlands/covenants/UnholyNova.tsx:67]: Fixed

**Review by DylanDirlam — APPROVED** [review]

**Pull request merged**